core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -139,8 +138,11 @@ class KafkaApis(val requestChannel: RequestChannel,
     request: RequestChannel.Request,
     handler: RequestChannel.Request => Unit
   ): Unit = {
-    def responseCallback(response: AbstractResponse): Unit = {
-      sendForwardedResponse(request, response)
+    def responseCallback(responseEither: Either[AbstractResponse, Errors]): Unit = {
+      responseEither match {
+        case Left(response) => sendForwardedResponse(request, response)
+        case Right(error) => closeConnection(request, Collections.singletonMap(error, 1))
```

Comment:
A debug message would be helpful here so that we know why the connection was closed. I think we might actually prefer to use `emptyMap()` here since the unsupported version error would be misleading.

Reply:
Sg, I prefer using `info` here since it should be a rare case.